File: src/types/order.ts
// Order management types matching Android app structure

export interface OrderItem {
  productId: string;
  name: string;
  sku: string;
  price: number;
  quantity: number;
  imageUrl: string;
  variantId: string;
  variantName: string;
  specifications: Record<string, string>;
}

export interface CustomerInfo {
  name: string;
  email: string;
  phone: string;
}

export interface ShippingAddress {
  recipientName: string;
  street: string;
  city: string;
  state: string;
  zipCode: string;
  country: string;
  phone: string;
  instructions: string;
}

export enum OrderStatus {
  PENDING = 'PENDING',
  CONFIRMED = 'CONFIRMED',
  PREPARING = 'PREPARING',
  READY = 'READY',
  SHIPPED = 'SHIPPED',
  DELIVERED = 'DELIVERED',
  CANCELLED = 'CANCELLED',
  REFUNDED = 'REFUNDED'
}

export enum PaymentStatus {
  PENDING = 'PENDING',
  PAID = 'PAID',
  FAILED = 'FAILED',
  REFUNDED = 'REFUNDED',
  PARTIAL_REFUND = 'PARTIAL_REFUND'
}

export enum FulfillmentStatus {
  PENDING = 'PENDING',
  PROCESSING = 'PROCESSING',
  SHIPPED = 'SHIPPED',
  DELIVERED = 'DELIVERED'
}

export enum ShippingMethod {
  PICKUP = 'PICKUP',
  DELIVERY = 'DELIVERY',
  SHIPPING = 'SHIPPING'
}

export interface Order {
  id: string;
  orderNumber: string;
  customerId: string;
  shopId: string;
  ownerId: string;
  
  // Order Items
  items: OrderItem[];
  
  // Pricing
  subtotal: number;
  tax: number;
  taxRate: number;
  shippingFee: number;
  deliveryFee: number;
  serviceFee: number;
  discount: number;
  discountCode: string;
  total: number;
  currency: string;
  
  // Status
  status: OrderStatus;
  paymentStatus: PaymentStatus;
  fulfillmentStatus: FulfillmentStatus;
  
  // Customer Info
  customerInfo: CustomerInfo;
  
  // Shipping
  shippingMethod: ShippingMethod;
  shippingAddress: ShippingAddress;
  trackingNumber: string;
  carrier: string;
  estimatedDelivery: number;
  
  // Payment
  paymentMethod: string;
  paymentReference: string;
  
  // Notes
  customerNotes: string;
  internalNotes: string;
  
  // Timestamps
  createdAt: number;
  updatedAt: number;
  confirmedAt: number;
  shippedAt: number;
  deliveredAt: number;
  cancelledAt: number;
}

export interface OrderOverview {
  pendingOrders: number;
  todaysOrders: number;
  shippedOrders: number;
  totalOrders: number;
}

// Helper function to get order status color and text
export const getOrderStatusInfo = (status: OrderStatus) => {
  switch (status) {
    case OrderStatus.PENDING:
      return { color: '#FF9800', text: 'Pending' };
    case OrderStatus.CONFIRMED:
      return { color: '#4CAF50', text: 'Confirmed' };
    case OrderStatus.PREPARING:
      return { color: '#9C27B0', text: 'Preparing' };
    case OrderStatus.READY:
      return { color: '#4CAF50', text: 'Ready' };
    case OrderStatus.SHIPPED:
      return { color: '#2196F3', text: 'Shipped' };
    case OrderStatus.DELIVERED:
      return { color: '#4CAF50', text: 'Delivered' };
    case OrderStatus.CANCELLED:
      return { color: '#F44336', text: 'Cancelled' };
    case OrderStatus.REFUNDED:
      return { color: '#F44336', text: 'Refunded' };
    default:
      return { color: '#9E9E9E', text: 'Unknown' };
  }
};

File: androidAPP/MyHustle/app/src/main/java/com/blueclipse/myhustle/data/model/Order.kt
package com.blueclipse.myhustle.data.model

/**
 * Represents an order in the MyHustle platform
 */
data class Order(
    val id: String = "",
    val orderNumber: String = "",
    val customerId: String = "",
    val shopId: String = "",
    val ownerId: String = "",
    
    // Order Items
    val items: List<OrderItem> = emptyList(),
    
    // Pricing
    val subtotal: Double = 0.0,
    val tax: Double = 0.0,
    val taxRate: Double = 0.0,
    val shippingFee: Double = 0.0,
    val deliveryFee: Double = 0.0,
    val serviceFee: Double = 0.0,
    val discount: Double = 0.0,
    val discountCode: String = "",
    val total: Double = 0.0,
    val currency: String = "USD",
    
    // Status
    val status: OrderStatus = OrderStatus.PENDING,
    val paymentStatus: PaymentStatus = PaymentStatus.PENDING,
    val fulfillmentStatus: FulfillmentStatus = FulfillmentStatus.PENDING,
    
    // Customer Info
    val customerInfo: CustomerInfo = CustomerInfo(),
    
    // Shipping
    val shippingMethod: ShippingMethod = ShippingMethod.PICKUP,
    val shippingAddress: ShippingAddress = ShippingAddress(),
    val trackingNumber: String = "",
    val carrier: String = "",
    val estimatedDelivery: Long = 0,
    
    // Payment
    val paymentMethod: String = "",
    val paymentReference: String = "",
    
    // Notes
    val customerNotes: String = "",
    val internalNotes: String = "",
    
    // Timestamps
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis(),
    val confirmedAt: Long = 0,
    val shippedAt: Long = 0,
    val deliveredAt: Long = 0,
    val cancelledAt: Long = 0
) {
    constructor() : this(
        "", "", "", "", "", emptyList(), 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, "", 0.0, "USD",
        OrderStatus.PENDING, PaymentStatus.PENDING, FulfillmentStatus.PENDING,
        CustomerInfo(), ShippingMethod.PICKUP, ShippingAddress(), "", "", 0,
        "", "", "", "", System.currentTimeMillis(), System.currentTimeMillis(), 0, 0, 0, 0
    )
}

/**
 * Order item details
 */
data class OrderItem(
    val productId: String = "",
    val name: String = "",
    val sku: String = "",
    val price: Double = 0.0,
    val quantity: Int = 1,
    val imageUrl: String = "",
    val variantId: String = "",
    val variantName: String = "",
    val specifications: Map<String, String> = emptyMap()
) {
    constructor() : this("", "", "", 0.0, 1, "", "", "", emptyMap())
}

/**
 * Customer information for orders
 */
data class CustomerInfo(
    val name: String = "",
    val email: String = "",
    val phone: String = ""
) {
    constructor() : this("", "", "")
}

/**
 * Shipping address information
 */
data class ShippingAddress(
    val recipientName: String = "",
    val street: String = "",
    val city: String = "",
    val state: String = "",
    val zipCode: String = "",
    val country: String = "",
    val phone: String = "",
    val instructions: String = ""
) {
    constructor() : this("", "", "", "", "", "", "", "")
}

/**
 * Order status enum
 */
enum class OrderStatus {
    PENDING,
    CONFIRMED,
    PREPARING,
    READY,
    SHIPPED,
    DELIVERED,
    CANCELLED,
    REFUNDED
}

/**
 * Payment status enum
 */
enum class PaymentStatus {
    PENDING,
    PAID,
    FAILED,
    REFUNDED,
    PARTIAL_REFUND
}

/**
 * Fulfillment status enum
 */
enum class FulfillmentStatus {
    PENDING,
    PROCESSING,
    SHIPPED,
    DELIVERED
}

/**
 * Shipping method enum
 */
enum class ShippingMethod {
    PICKUP,
    DELIVERY,
    SHIPPING
}
